tests/benchmark/utils/s2n_negotiate_server_benchmark.cc

```diff
@@ -0,0 +1,304 @@
+    const char *cipher_prefs = "test_all_tls12";
+
+    const char *certificates[MAX_CERTIFICATES] = {0};
+    const char *private_keys[MAX_CERTIFICATES] = {0};
+
+
+    int parallelize = 0;
+
+    conn_settings.session_ticket = 1;
+    conn_settings.session_cache = 0;
+    conn_settings.max_conns = -1;
+    conn_settings.psk_list_len = 0;
+    conn_settings.insecure = insecure;
+
+    conn_settings.use_corked_io = use_corked_io;
+
+    int max_early_data = 0;
+
+    s2n_init();
+
+    if (conn_settings.prefer_throughput && conn_settings.prefer_low_latency) {
+        fprintf(stderr, "prefer-throughput and prefer-low-latency options are mutually exclusive\n");
+        exit(1);
+    }
+
+    GUARD_EXIT(setvbuf(stdin, NULL, _IONBF, 0), "Error disabling buffering for stdin\n");
+    GUARD_EXIT(setvbuf(stdout, NULL, _IONBF, 0), "Error disabling buffering for stdout\n");
+
+    memset(&hints, 0, sizeof(hints));
+
+    hints.ai_family = AF_UNSPEC;
+    hints.ai_socktype = SOCK_STREAM;
+
+    if (signal(SIGPIPE, SIG_IGN) == SIG_ERR) {
+        fprintf(stderr, "Error disabling SIGPIPE\n");
+        exit(1);
+    }
+
+    GUARD_EXIT(getaddrinfo(host, port, &hints, &ai), "getaddrinfo error\n");
+    GUARD_EXIT((sockfd = socket(ai->ai_family, ai->ai_socktype, ai->ai_protocol)), "socket error\n");
+    GUARD_EXIT(setsockopt(sockfd, SOL_SOCKET, SO_REUSEADDR, &r, sizeof(int)), "setsockopt error\n");
+    GUARD_EXIT(bind(sockfd, ai->ai_addr, ai->ai_addrlen), "bind error");
+    GUARD_EXIT(listen(sockfd, 1), "listen error\n");
+
+
+    if (DEBUG_PRINT) {
+        printf("Listening on %s:%s\n", host, port);
+    }
+
+    config_once = s2n_config_new();
+    if (!config_once) {
+        print_s2n_error("Error getting new s2n config");
+        exit(1);
+    }
+
+    s2n_set_common_server_config(max_early_data, config_once, conn_settings, cipher_prefs, session_ticket_key_file_path);
+
+
+    certificates[0] = rsa_certificate_chain;
+    private_keys[0] = rsa_private_key;
+    struct s2n_cert_chain_and_key *chain_and_key_rsa = s2n_cert_chain_and_key_new();
+    GUARD_EXIT(s2n_cert_chain_and_key_load_pem(chain_and_key_rsa, certificates[0], private_keys[0]),
+               "Error getting certificate/key");
+
+    GUARD_EXIT(s2n_config_add_cert_chain_and_key_to_store(config_once, chain_and_key_rsa),
+               "Error setting certificate/key");
+
+    certificates[0] = ecdsa_certificate_chain;
+    private_keys[0] = ecdsa_private_key;
+
+    struct s2n_cert_chain_and_key *chain_and_key = s2n_cert_chain_and_key_new();
+    GUARD_EXIT(s2n_cert_chain_and_key_load_pem(chain_and_key, certificates[0], private_keys[0]),
+               "Error getting certificate/key");
+
+    GUARD_EXIT(s2n_config_add_cert_chain_and_key_to_store(config_once, chain_and_key),
+               "Error setting certificate/key");
+
+    bool stop_listen = false;
+    while ((!stop_listen) && (fd_bench = accept(sockfd, ai->ai_addr, &ai->ai_addrlen)) > 0) {
+        if (!parallelize) {
```

Comment:
What happens if this is true? Also do we need this? It looks like this is always false.